src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/SyntaxNodeExtensions.cs

```diff
@@ -15,10 +16,15 @@ internal static class SyntaxNodeExtensions
 
         public static SyntaxNode AddMemberAttributes(this SyntaxNode node,
             SyntaxGenerator syntaxGenerator,
+            ISymbol member,
             ISymbolFilter symbolFilter,
-            ISymbol member)
+            ISymbolFilter? attributeDataSymbolFilter)
         {
-            foreach (AttributeData attribute in member.GetAttributes().ExcludeNonVisibleOutsideOfAssembly(symbolFilter))
+            ImmutableArray<AttributeData> memberAttributes = member.GetAttributes()
+                .ExcludeWithFilter(attributeDataSymbolFilter)
```

Comment:
Same feedback as the other PR - we can combine filters here to save double filtering.